test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+import { ETH } from "contracts/src/libraries/Constants.sol";
+import { FixedPointMath, ONE } from "contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveTest } from "test/utils/HyperdriveTest.sol";
+import { HyperdriveUtils } from "test/utils/HyperdriveUtils.sol";
+import { Lib } from "test/utils/Lib.sol";
+
+/// @author DELV
+/// @title IntegrationTest
+/// @notice The base contract for the integration testing suite.
+/// @dev A testing suite that provides a foundation to setup, deploy, and
+/// test common cases for Hyperdrive instances.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract IntegrationTest is HyperdriveTest {
+    using Lib for *;
+    using FixedPointMath for uint256;
+
+    // Fixed rate used to configure market.
+    uint256 internal constant FIXED_RATE = 0.05e18;
+
+    // ETH whale account.
+    address internal ETH_WHALE = 0x00000000219ab540356cBB839Cbe05303d7705Fa;
+
+    /// @dev Configuration for the Integration testing suite.
+    struct IntegrationConfig {
+        address[] whaleAccounts;
+        IERC20 token;
+        IERC20 baseToken;
+        uint256 shareTolerance;
+        uint256 minTransactionAmount;
+        uint256 positionDuration;
+        bool enableBaseDeposits;
+    }
+
+    IntegrationConfig internal config;
+    IHyperdrive.PoolDeployConfig internal poolConfig;
+    HyperdriveFactory factory;
+    address deployerCoordinator;
+    bool internal immutable isBaseETH;
+
+    /// @dev Constructor for the Integration testing suite.
+    /// @param _config The Integration configuration.
+    constructor(IntegrationConfig storage _config) {
+        config = _config;
+        isBaseETH = config.baseToken == IERC20(ETH);
+    }
+
+    /// @notice Forge setup function.
+    /// @dev Inherits from HyperdriveTest and deploys the
+    ///      Hyperdrive factory, deployer coordinator, and targets.
+    function setUp() public virtual override {
+        super.setUp();
+
+        // Fund accounts with ETH and token from whales.
+        vm.deal(alice, 100_000e18);
+        vm.deal(bob, 100_000e18);
+        address[] memory accounts = new address[](2);
+        accounts[0] = alice;
+        accounts[1] = bob;
+        for (uint256 i = 0; i < config.whaleAccounts.length; i++) {
+            fundAccounts(
+                address(hyperdrive),
+                config.token,
+                config.whaleAccounts[i],
+                accounts
+            );
+        }
+
+        // Deploy the Hyperdrive Factory contract.
+        deployFactory();
+
+        // Set the deployer coordinator address and add to the factory.
+        deployerCoordinator = deployCoordinator();
+        factory.addDeployerCoordinator(deployerCoordinator);
+
+        // Deploy all Hyperdrive targets using  deployer coordinator contract.
```

Comment:
This comment is misleading since we also deploy the instance. I'm slightly confused about why we are deploying this considering that we don't do anything with the return value.

Reply:
It feels like this is mostly done for historical reasons, and unless it's doing something that I'm not picking up on, we could consider getting rid of it.